cfme/intelligence/reports/reports.py

```diff
@@ -386,38 +386,45 @@ def delete(self, cancel=False):
     def saved_reports(self):
         return self.collections.saved_reports
 
-    def create_schedule(self, name=None, description=None, active=None,
-            timer=None, from_email=None, emails=None, email_options=None):
-
+    def create_schedule(
+        self,
+        name=None,
+        description=None,
+        active=True,
+        timer=None,
+        email=None,
+        email_options=None,
+        cancel=False,
+    ):
         view = navigate_to(self, "ScheduleReport")
-        view.fill({
+        if email:
+            email["emails_send"] = True
+
+        data = {
             "name": name,
             "description": description,
             "active": active,
-            "run": timer.get("run"),
-            "time_zone": timer.get("time_zone"),
-            "starting_date": timer.get("starting_date"),
-            "hour": timer.get("hour"),
-            "minute": timer.get("minute"),
-            "emails_send": bool(emails),
-            "from_email": from_email,
-            "emails": emails,
-            "send_if_empty": email_options.get("send_if_empty"),
-            "send_txt": email_options.get("send_txt"),
-            "send_csv": email_options.get("send_csv"),
-            "send_pdf": email_options.get("send_pdf")
-        })
+            "report_filter": {
+                "filter_type": self.company_name,
+                "subfilter_type": self.subtype,
+                "report_type": self.menu_name,
+            },
+            "timer": timer,
+            "email": email,
+            "email_options": email_options,
+        }
```

Comment:
Looking good @valaparthvi .
Optional : the data part can be done in fill_dict method to be consistent with some other code .
Example : https://github.com/ManageIQ/integration_tests/blob/master/cfme/v2v/conversion_hosts.py#L142

Reply:
@sshveta implemented those changes.